components/table/hooks/useFilter/index.tsx

```diff
@@ -207,11 +209,30 @@ function useFilter<RecordType>({
   const mergedFilterStates = React.useMemo(() => {
     const collectedStates = collectFilterStates(mergedColumns, false);
 
+    const filteredKeysIsNotControlled = collectedStates.every(
+      ({ filteredKeys }) => filteredKeys === undefined,
+    );
```

Comment:
Could we just:
```tsx
devWarning(
  filteredKeysIsNotControlled || collectedStates.every(({ filteredKeys }) => filteredKeys !== undefined),
  ...
);
```

We just need to warning that developer not fully provide the control value.

Reply:
Ok.